vignettes/easier_manual.Rmd

```diff
@@ -18,123 +18,218 @@ knitr::opts_chunk$set(
 
-# ******************
-# Example: Riaz
-# ******************
-data("Riaz_data")
-
-# ---------------------------------------------------------------------------
-# Computation of cell fractions
-# ---------------------------------------------------------------------------
-cell_fractions <- compute_cell_fractions(RNA.tpm = Riaz_data$tpm_RNAseq)
-
-# ---------------------------------------------------------------------------
-# Computation of pathways scores
-# ---------------------------------------------------------------------------
-Pathway_activities <- compute_pathways_scores(RNA.counts = Riaz_data$raw_counts_RNAseq,
-                                              remove.genes.ICB_proxies = TRUE)
-
-# ---------------------------------------------------------------------------
-# Computation of TF activity
-# ---------------------------------------------------------------------------
-TF_activities <- compute_TF_activity(RNA.tpm = Riaz_data$tpm_RNAseq,
-                                     remove.genes.ICB_proxies = FALSE)
-
-# ---------------------------------------------------------------------------
-# Computation of LR pairs weights
-# ---------------------------------------------------------------------------
-LR_data <- compute_LR_pairs(RNA.tpm = Riaz_data$tpm_RNAseq,
-                            remove.genes.ICB_proxies = FALSE,
-                            compute.cytokines.pairs = FALSE,
-                            cancertype = cancertype)
-
-# ---------------------------------------------------------------------------
-# Computation of cell-cell interactions scores
-# ---------------------------------------------------------------------------
-CCpairs_grouped <- compute_CC_pairs_grouped(lrpairs = LR_data$LRpairs,
-                                            cancertype = cancertype)
-
-# ---------------------------------------------------------------------------
-# Predict immune response
-# ---------------------------------------------------------------------------
-
-# *********************
-# Collect views data
-pathways_scores <- Pathway_activities$scores
-cell_fractions <- cell_fractions
-lrpairs_weights <- LR_data$LRpairs
-tfs_scores <- TF_activities$scores
-ccpairsgrouped_scores <- CCpairs_grouped$score
-
-# *********************
-# Specify cancer type
-cancertype <- "SKCM"
-
-predictions_immune_response <- predict_immune_response(pathways = pathways_scores,
+```{r eval = FALSE}
+# Create a external dataset object
+setClass("DatasetUser", slots=list(name="character", counts="data.frame", tpm="data.frame", response="character", TMB="numeric", cancertype="character"))
+
+obj_dataset <- new("DatasetUser",
+                   name="Mariathasan", 
+                   counts=mariathasan_data$counts, 
+                   tpm=mariathasan_data$tpm, 
+                   response=patient_response, 
+                   TMB=TumorMutationalBurden,
+                   cancertype="BLCA"
+                   )
```

Comment:
I am not sure this is the best way to do it.

I mean - yes, a class is great to define its validity and so, but:
- should not belong to the vignette, but rather to the package itself
- we should also define proper accessors to avoid users reading into `object@slot` directly, as we do later

Maybe a simple list would do it?

Re: name of the class, it is a little too generic - maybe something that "fits" the fact it is used in the context of this package?